openjdk8/src/main/docker/Dockerfile

```diff
@@ -30,12 +30,19 @@ RUN echo 'deb http://httpredir.debian.org/debian jessie-backports main' > /etc/a
 # workaround for https://bugs.debian.org/775775
 RUN /var/lib/dpkg/info/ca-certificates-java.postinst configure
 
-# Upgrade to OpenSSL 1.0.2
-ADD http://ftp.us.debian.org/debian/pool/main/o/openssl/libssl1.0.2_1.0.2f-2_amd64.deb /tmp/
-ADD http://ftp.us.debian.org/debian/pool/main/o/openssl/openssl_1.0.2f-2_amd64.deb /tmp/
-RUN dpkg -i /tmp/libssl1.0.2_1.0.2f-2_amd64.deb /tmp/openssl_1.0.2f-2_amd64.deb \
- && rm /tmp/libssl1.0.2_1.0.2f-2_amd64.deb /tmp/openssl_1.0.2f-2_amd64.deb \
- && apt-get -y -q upgrade \
+# Upgrade to OpenSSL 1.0.2 (via sid)
+ADD sid.list /etc/apt/sources.list.d/
+RUN apt-get -y update \
+ && apt-get -y -q --no-install-recommends install \
+    libssl1.0.2 \
+    openssl
+
+# Cleanup sid references
+RUN rm /etc/apt/sources.list.d/sid.list \
+ && apt-get -y update
+
+# Cleanup apt-get temporary files
+RUN apt-get -y -q upgrade \
```

Comment:
Could we merge all the runs into a single Docker run step to minimize intermediate layers?